src/dmd/printast.d

```diff
@@ -57,7 +57,7 @@ extern (C++) final class PrintASTVisitor : Visitor
     override void visit(RealExp e)
     {
         printIndent(indent);
-        printf("Real %Lg %s\n", e.value, e.type ? e.type.toChars() : "");
+        printf("Real %Lg %s\n", cast(real) e.value, e.type ? e.type.toChars() : "");
```

Comment:
You can't cast a soft float type to a native, one. We have CTFloat to handle the sprintf for that.

Reply:
`longdouble_soft` implements [`opCast!real`]?(https://github.com/dlang/dmd/blob/32313b9f696520fae4910b96d5e4e6bd9b28e4a8/src/dmd/root/longdouble.d#L177-L184)?

So this should be `CTFloat.sprint` + a temporary buffer?

Reply:
> `longdouble_soft` implements [`opCast!real`]?

That's a pseudo-soft float type, not like [gcc](https://github.com/D-Programming-GDC/gcc/blob/4d46018791ee0331b2d867b2420ca566ea4c3a8f/gcc/real.h#L39-L57) that bears no resemblance to anything native.

> So this should be `CTFloat.sprint` + a temporary buffer?

Yes.